packages/react/src/ImageSlider/ImageSlider.tsx

```diff
@@ -111,11 +173,11 @@ export const ImageSliderRoot = forwardRef(
 
     const goToPreviousSlide = () => {
```

Comment:
Can some or all of these functions be moved outside of the component? They keep getting redefined like this. Some will need an extra parameter.

Reply:
The observers are in `useEffect` to prevent rerenders. And alle these functions are in here to prevent juggling that ref around. I'm not sure how `useMemo` would help. This is to cache results between re-renders?

The component would have been much cleaner in vanilla javascript, "react-izing" caused a lot of headache so I'm open to suggestions.